library/kani/src/futures.rs

```diff
@@ -41,3 +44,216 @@ const NOOP_RAW_WAKER: RawWaker = {
 
     RawWaker::new(std::ptr::null(), &RawWakerVTable::new(clone_waker, noop, noop, noop))
 };
+
+static mut EXECUTOR: Scheduler = Scheduler::new();
+const MAX_TASKS: usize = 16;
+
+type BoxFuture = Pin<Box<dyn Future<Output = ()> + Sync + 'static>>;
+
+/// Allows to parameterize how the scheduler picks the next task to poll in `spawnable_block_on`
+pub trait SchedulingStrategy {
+    /// Picks the next task to be scheduled whenever the scheduler needs to pick a task to run next, and whether it can be assumed that the picked task is still running
```

Comment:
It's not clear to me what the second parameter means here. What kind of information the scheduler has to make that decision in the first place?

Reply:
I'm sorry, I'm not quite sure what you're asking here. The second parameter is the number of tasks. Are you talking about the second part of the return value? This is just a flag indicating whether the scheduler can `kani::assume` that the returned task is still running, usually because the task was chosen nondeterministically. This cuts off "useless" paths where a completed task is polled again, and is even necessary to make things terminate, if nondeterminism is involved: if we pick the task nondeterministically, and don't have the restriction to still running tasks, we could poll the same task over and over again. However, for the round robin scheduling strategy, assuming that the picked task is still running is generally not possible, and amounts to `assume(false)`, which is undesirable.

Reply:
Sorry, I was talking about the return value. What does the `bool` part of the return value mean?

Reply:
This is just a flag indicating whether the scheduler can `kani::assume` that the returned task is still running. As a rule of thumb: if the scheduling strategy picks the next task nondeterministically (using `kani::any()`), return `true`, otherwise false. If true, the scheduler will then assume that the picked task is still running, which cuts off "useless" paths where a completed task is polled again, and is even necessary to make things terminate, if nondeterminism is involved: if we pick the task nondeterministically, and don't have the restriction to still running tasks, we could poll the same task over and over again. However, for most deterministic scheduling strategies, e.g. the round robin scheduling strategy, assuming that the picked task is still running is generally not possible because if that task has ended, we are saying `assume(false)` and the verification effectively stops (which is undesirable, of course).

Reply:
I think you should document that. :)

Reply:
Updated the docs.